### ISSUE TITLE
kube-state-metrics serviceaccount installed to kube-system namespace by default

### ISSUE THREAD

**Issue by devurandom** [digitalocean/DOKS#25]
It appears that in a fresh DigitalOcean Kubernetes cluster the `kube-system` namespace already contains a `kube-state-metrics` serviceaccount:
```
apiVersion: v1
kind: ServiceAccount
metadata:
  annotations:
    kubectl.kubernetes.io/last-applied-configuration: |
      {"apiVersion":"v1","imagePullSecrets":[],"kind":"ServiceAccount","metadata":{"annotations":{},"name":"kube-state-metrics","namespace":"kube-system"}}
  creationTimestamp: "2021-04-24T16:33:13Z"
  managedFields:
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:secrets:
        .: {}
        k:{"name":"kube-state-metrics-token-766fc"}:
          .: {}
          f:name: {}
    manager: kube-controller-manager
    operation: Update
    time: "2021-04-24T16:33:13Z"
  - apiVersion: v1
    fieldsType: FieldsV1
    fieldsV1:
      f:metadata:
        f:annotations:
          .: {}
          f:kubectl.kubernetes.io/last-applied-configuration: {}
    manager: kubectl-client-side-apply
    operation: Update
    time: "2021-04-24T16:33:13Z"
  name: kube-state-metrics
  namespace: kube-system
  resourceVersion: "288"
  uid: 1e228525-a73b-4a93-83d2-09c06ca719b8
secrets:
- name: kube-state-metrics-token-766fc
```

This is even though kube-state-metrics was not installed (and is not running in any namespace).

The cluster is at Kubernetes version `1.20.2-do.0` and was created using the [Terraform DigitalOcean provider](https://github.com/hashicorp/terraform-provider-digitalocean) version `2.7.0`.

This trips the [kube-state-metrics Helm chart](https://github.com/kubernetes/kube-state-metrics/tree/master/charts/kube-state-metrics) version `2.13.2`, which also tries to create this serviceaccount:

> Error: rendered manifests contain a resource that already exists. Unable to continue with install: ServiceAccount "kube-state-metrics" in namespace "kube-system" exists and cannot be imported into the current release: invalid ownership metadata; label validation error: missing key "app.kubernetes.io/managed-by": must be set to "Helm"; annotation validation error: missing key "meta.helm.sh/release-name": must be set to "kube-state-metrics"; annotation validation error: missing key "meta.helm.sh/release-namespace": must be set to "kube-system"

In a cluster created a long time ago with Kubernetes 1.18 and using a different version of the kube-state-metrics Helm chart this was not the case.

Is it possible to not create this serviceaccount automatically?

**Comment by devurandom** [comment]
Not only a service account exists already, but also the clusterrole and clusterrolebinding, which both trip up the Helm chart.

**Comment by devurandom** [comment]
Something is quite fishy. I installed kube-state-metrics using the Helm chart as explained above, with `values.yaml`:

```yaml
serviceAccount:
  create: false
rbac:
  create: false
```

But I get lots and lots of errors (example):
```
E0427 10:23:14.724801       1 reflector.go:156] pkg/mod/k8s.io/client-go@v0.0.0-20191109102209-3c0d1af94be5/tools/cache/reflector.go:108: Failed to list *v1.DaemonSet: daemonsets.apps is forbidden: User "system:serviceaccount:kube-system:default" cannot list resource "daemonsets" in API group "apps" at the cluster scope
```

So somehow the service account DigitalOcean installs by default does not have sufficient permissions to actually run kube-state-metrics?

**Comment by ianmuscat** [comment]
Using the following Helm chart values worked around the issue for me.
```
serviceAccount:
  create: false
  name: kube-state-metrics
rbac:
  create: false
```

**Comment by timoreimann** [comment]
Apologies for the late note, we fixed this in 1.21.2-do.0 and 1.20.8-do.0, respectively. We do not delete the prior policy files automatically, however, so affected customers will need to delete/overwrite them on their own.
